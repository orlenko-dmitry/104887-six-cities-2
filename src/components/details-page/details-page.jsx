import React, {PureComponent} from 'react';
import {
  arrayOf,
  shape,
  string,
  number,
  bool,
  func,
} from 'prop-types';
import {connect} from 'react-redux';

import {defineRating, getNearOffers} from '../../helpers/helpers.js';
import ReviewsList from '../reviews-list/reviews-list.jsx';
import OffersMap from '../offers-map/offers-map.jsx';
import OffersList from '../offers-list/offers-list.jsx';
import ReviewsFrom from '../reviews-form/reviews-form.jsx';
import withReviewsForm from '../../hocs/with-reviews-form/with-reviews-form.jsx';
import aFilters from '../../store/filters/actions.js';
import aData from '../../store/data/actions.js';
import {getCityOffers} from '../../store/data/selectors.js';

const WithReviewsForm = withReviewsForm(ReviewsFrom);

class DetailsPage extends PureComponent {
  constructor(props) {
    super(props);
    this.colorPinHandler = this.colorPinHandler.bind(this);
  }

  componentDidMount() {
    const {fetchCommentsHandler, match: {params: {offerId}}} = this.props;
    fetchCommentsHandler(offerId);
  }

  colorPinHandler(id) {
    const {getOfferIdHandler} = this.props;
    getOfferIdHandler(id);
  }

  render() {
    const {
      offers,
      comments,
      city,
      onHoverOfferId,
      match: {
        params: {
          offerId,
        }
      }
    } = this.props;

    const offerIndex = offers.map((offer) => offer.id).indexOf(Number(offerId));

    const {
      images,
      isPremium,
      title,
      rating,
      price,
      bedrooms,
      maxAdults,
      description,
      id,
      goods,
      host: {
        isPro,
        name: hostName,
        avatarUrl,
      },
    } = offers[offerIndex];
    return offers.length && (
      <main className="page__main page__main--property">
        <section className="property">
          <div className="property__gallery-container container">
            <div className="property__gallery">
              {images.map((image, index) => (
                <div className="property__image-wrapper" key={index}>
                  <img className="property__image" src={image} alt="Photo studio" />
                </div>
              ))}
            </div>
          </div>
          <div className="property__container container">
            <div className="property__wrapper">
              {isPremium && (
                <div className="property__mark">
                  <span>Premium</span>
                </div>
              )}
              <div className="property__name-wrapper">
                <h1 className="property__name">
                  {title}
                </h1>
                <button className="property__bookmark-button button" type="button">
                  <svg className="property__bookmark-icon" width={31} height={33}>
                    <use xlinkHref="#icon-bookmark" />
                  </svg>
                  <span className="visually-hidden">To bookmarks</span>
                </button>
              </div>
              <div className="property__rating rating">
                <div className="property__stars rating__stars">
                  <span style={{width: defineRating(rating)}} />
                  <span className="visually-hidden">Rating</span>
                </div>
                <span className="property__rating-value rating__value">{rating}</span>
              </div>
              <ul className="property__features">
                <li className="property__feature property__feature--entire">
                  Entire place
                </li>
                <li className="property__feature property__feature--bedrooms">
                  {bedrooms} Bedrooms
                </li>
                <li className="property__feature property__feature--adults">
                  Max {maxAdults} adults
                </li>
              </ul>
              <div className="property__price">
                <b className="property__price-value">â‚¬{price}</b>
                <span className="property__price-text">&nbsp;night</span>
              </div>
              <div className="property__inside">
                <h2 className="property__inside-title">What&apos;s inside</h2>
                <ul className="property__inside-list">
                  {goods.map((good, index) => (
                    <li className="property__inside-item" key={index}>
                      {good}
                    </li>
                  ))}
                </ul>
              </div>
              <div className="property__host">
                <h2 className="property__host-title">Meet the host</h2>
                <div className="property__host-user user">
                  <div className="property__avatar-wrapper property__avatar-wrapper--pro user__avatar-wrapper">
                    <img className="property__avatar user__avatar" src={avatarUrl} width={74} height={74} alt="Host avatar" />
                  </div>
                  <span className="property__user-name">
                    {hostName}
                  </span>
                  {isPro && (
                    <span className="property__user-status">
                      Pro
                    </span>
                  )}
                </div>
                <div className="property__description">
                  <p className="property__text">
                    {description}
                  </p>
                </div>
              </div>
              <section className="property__reviews reviews">
                <h2 className="reviews__title">Reviews <span className="reviews__amount">{comments.length}</span></h2>
                <ReviewsList reviews={comments} />
                <WithReviewsForm />
              </section>
            </div>
          </div>
          <section className="property__map map">
            <OffersMap
              offers={getNearOffers(offers, id, true)}
              selectedCity={city}
              selectedOfferId={id}
              onHoverOfferId={onHoverOfferId}
            />
          </section>
        </section>
        <div className="container">
          <section className="near-places places">
            <h2 className="near-places__title">Other places in the neighbourhood</h2>
            <OffersList
              classNames={`near-places__list`}
              offers={getNearOffers(offers, id)}
              onColorPin={this.colorPinHandler}
            />
          </section>
        </div>
      </main>
    );
  }
}

DetailsPage.propTypes = {
  offers: arrayOf(shape({
    id: number.isRequired,
    images: arrayOf(string).isRequired,
    title: string.isRequired,
    isFavorite: bool.isRequired,
    isPremium: bool.isRequired,
    rating: number.isRequired,
    bedrooms: number.isRequired,
    maxAdults: number.isRequired,
    price: number.isRequired,
    goods: arrayOf(string).isRequired,
    description: string.isRequired,
    host: shape({
      id: number.isRequired,
      isPro: bool.isRequired,
      name: string.isRequired,
      avatarUrl: string.isRequired,
    }),
  })).isRequired,
  comments: arrayOf(shape({})).isRequired,
  city: shape({
    location: shape({
      latitude: number.isRequired,
      longitude: number.isRequired,
      zoom: number.isRequired,
    }).isRequired,
    name: string.isRequired,
  }).isRequired,
  match: shape({
    params: shape({
      offerId: string.isRequired,
    }).isRequired
  }).isRequired,
  onHoverOfferId: number.isRequired,
  getOfferIdHandler: func.isRequired,
  fetchCommentsHandler: func.isRequired,
};

const mapStateToProps = ({rData, rFilters}) => ({
  offers: getCityOffers({rData, rFilters}),
  city: rData.city,
  comments: rData.comments,
  onHoverOfferId: rFilters.onHoverOfferId,
});

const mapDispatchToProps = (dispatch) => ({
  getOfferIdHandler: (payload) => dispatch(aFilters.getOfferId(payload)),
  fetchCommentsHandler: (payload) => dispatch(aData.fetchComments(payload)),
});

export {DetailsPage};

export default connect(mapStateToProps, mapDispatchToProps)(DetailsPage);
