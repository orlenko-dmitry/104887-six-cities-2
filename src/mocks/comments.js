export default ([
  {
    comment: `The deluxe room was a quite comfortable one with all the adequate facilities. The only thing that made me feel uncomfortable was the rude behavior of an impolite staff at the reception desk.`,
    date: `2019-12-02T13:10:30.195Z`,
    id: 1,
    rating: 3,
    user: {
      avatarUrl: `https://htmlacademy-react-2.appspot.com/six-cities/static/avatar/10.jpg`,
      id: 19,
      isPro: false,
      name: `Christina`,
    }
  },
  {
    comment: `I stayed here for one night and it was an unpleasant experience.`,
    date: `2019-12-02T13:10:30.195Z`,
    id: 2,
    rating: 2,
    user: {
      avatarUrl: `https://htmlacademy-react-2.appspot.com/six-cities/static/avatar/2.jpg`,
      id: 11,
      isPro: false,
      name: `Jack`,
    }
  },
]);
