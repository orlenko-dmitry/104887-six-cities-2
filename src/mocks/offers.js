const offers = [{
  "id": 1,
  "city": {
    "name": `Amsterdam`,
    "location": {
      "latitude": 52.370216,
      "longitude": 4.895168,
      "zoom": 10
    }
  },
  "previewImage": `img/apartment-01.jpg`,
  "images": [`/img/room.jpg`, `img/apartment-01.jpg`, `img/apartment-02.jpg`, `img/apartment-03.jpg`],
  "title": `Beautiful & luxurious studio at great location`,
  "isFavorite": true,
  "isPremium": true,
  "rating": 4.8,
  "type": `apartment`,
  "bedrooms": 3,
  "max_adults": 4,
  "price": 120,
  "goods": [`Heating`, `Kitchen`, `Cable TV`, `Washing machine`, `Coffee machine`, `Dishwasher`],
  "host": {
    "id": 3,
    "isPro": true,
    "name": `Angelina`,
    "avatar_url": `img/avatar-angelina.jpg`
  },
  "description": `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
  "location": {
    "latitude": 52.35514938496378,
    "longitude": 4.673877537499948,
    "zoom": 8
  }
}, {
  "id": 2,
  "city": {
    "name": `Amsterdam`,
    "location": {
      "latitude": 52.370216,
      "longitude": 4.895168,
      "zoom": 10
    }
  },
  "previewImage": `img/room.jpg`,
  "images": [`/img/room.jpg`, `img/apartment-01.jpg`, `img/apartment-02.jpg`, `img/apartment-03.jpg`],
  "title": `Wood and stone place`,
  "isFavorite": false,
  "isPremium": false,
  "rating": 4.2,
  "type": `private room`,
  "bedrooms": 1,
  "max_adults": 2,
  "price": 80,
  "goods": [`Heating`, `Kitchen`, `Cable TV`, `Washing machine`, `Coffee machine`, `Dishwasher`],
  "host": {
    "id": 3,
    "isPro": true,
    "name": `Angelina`,
    "avatar_url": `img/avatar-angelina.jpg`
  },
  "description": `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
  "location": {
    "latitude": 52.35514938496378,
    "longitude": 4.673877537499948,
    "zoom": 8
  }
}, {
  "id": 3,
  "city": {
    "name": `Amsterdam`,
    "location": {
      "latitude": 52.370216,
      "longitude": 4.895168,
      "zoom": 10
    }
  },
  "previewImage": `img/apartment-02.jpg`,
  "images": [`/img/room.jpg`, `img/apartment-01.jpg`, `img/apartment-02.jpg`, `img/apartment-03.jpg`],
  "title": `Nice, cozy, warm big bed apartmen`,
  "isFavorite": true,
  "isPremium": false,
  "rating": 4,
  "type": `apartment`,
  "bedrooms": 2,
  "max_adults": 3,
  "price": 70,
  "goods": [`Heating`, `Kitchen`, `Cable TV`, `Washing machine`, `Coffee machine`, `Dishwasher`],
  "host": {
    "id": 3,
    "isPro": true,
    "name": `Angelina`,
    "avatar_url": `img/avatar-angelina.jpg`
  },
  "description": `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
  "location": {
    "latitude": 52.35514938496378,
    "longitude": 4.673877537499948,
    "zoom": 8
  }
}, {
  "id": 4,
  "city": {
    "name": `Amsterdam`,
    "location": {
      "latitude": 52.370216,
      "longitude": 4.895168,
      "zoom": 10
    }
  },
  "previewImage": `img/apartment-03.jpg`,
  "images": [`/img/room.jpg`, `img/apartment-01.jpg`, `img/apartment-02.jpg`, `img/apartment-03.jpg`],
  "title": `Beautiful luxurious apartment at great location`,
  "isFavorite": false,
  "isPremium": true,
  "rating": 4.9,
  "type": `apartment`,
  "bedrooms": 3,
  "max_adults": 6,
  "price": 200,
  "goods": [`Heating`, `Kitchen`, `Cable TV`, `Washing machine`, `Coffee machine`, `Dishwasher`],
  "host": {
    "id": 3,
    "isPro": true,
    "name": `Angelina`,
    "avatar_url": `img/avatar-angelina.jpg`
  },
  "description": `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
  "location": {
    "latitude": 52.35514938496378,
    "longitude": 4.673877537499948,
    "zoom": 8
  }
}];

export default offers;
