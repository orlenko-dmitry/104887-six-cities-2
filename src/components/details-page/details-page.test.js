import React from 'react';
import {shallow} from 'enzyme';

import {DetailsPage} from '../details-page/details-page.jsx';
import offers from '../../mocks/offers.js';
import comments from '../../mocks/comments.js';
import {ASYNC_STATUSES} from '../../consts/index.js';
const city = offers[0].city;
const match = {
  params: {
    offerId: `3`,
  },
};

it(`DetailsPage renders correctly`, () => {
  const tree = shallow(
      <DetailsPage
        offers={offers}
        comments={comments}
        city={city}
        user={null}
        onHoverOfferId={-1}
        offersFetchStatus={ASYNC_STATUSES.SUCCESS}
        match={match}
        getOfferIdHandler={() => {}}
        fetchCommentsHandler={() => {}}
        favoriteAddHandler={() => {}}
      />
  );

  expect(tree).toMatchSnapshot();
});
