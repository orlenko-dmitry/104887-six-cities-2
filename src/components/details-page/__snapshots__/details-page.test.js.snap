// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailsPage renders correctly 1`] = `
<main
  className="page__main page__main--property"
>
  <section
    className="property"
  >
    <div
      className="property__gallery-container container"
    >
      <div
        className="property__gallery"
      >
        <div
          className="property__image-wrapper"
          key="0"
        >
          <img
            alt="Photo studio"
            className="property__image"
            src="https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/3.jpg"
          />
        </div>
        <div
          className="property__image-wrapper"
          key="1"
        >
          <img
            alt="Photo studio"
            className="property__image"
            src="https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/1.jpg"
          />
        </div>
        <div
          className="property__image-wrapper"
          key="2"
        >
          <img
            alt="Photo studio"
            className="property__image"
            src="https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/9.jpg"
          />
        </div>
        <div
          className="property__image-wrapper"
          key="3"
        >
          <img
            alt="Photo studio"
            className="property__image"
            src="https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/19.jpg"
          />
        </div>
        <div
          className="property__image-wrapper"
          key="4"
        >
          <img
            alt="Photo studio"
            className="property__image"
            src="https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/4.jpg"
          />
        </div>
        <div
          className="property__image-wrapper"
          key="5"
        >
          <img
            alt="Photo studio"
            className="property__image"
            src="https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/10.jpg"
          />
        </div>
        <div
          className="property__image-wrapper"
          key="6"
        >
          <img
            alt="Photo studio"
            className="property__image"
            src="https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/6.jpg"
          />
        </div>
        <div
          className="property__image-wrapper"
          key="7"
        >
          <img
            alt="Photo studio"
            className="property__image"
            src="https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/15.jpg"
          />
        </div>
        <div
          className="property__image-wrapper"
          key="8"
        >
          <img
            alt="Photo studio"
            className="property__image"
            src="https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/12.jpg"
          />
        </div>
        <div
          className="property__image-wrapper"
          key="9"
        >
          <img
            alt="Photo studio"
            className="property__image"
            src="https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/7.jpg"
          />
        </div>
        <div
          className="property__image-wrapper"
          key="10"
        >
          <img
            alt="Photo studio"
            className="property__image"
            src="https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/2.jpg"
          />
        </div>
        <div
          className="property__image-wrapper"
          key="11"
        >
          <img
            alt="Photo studio"
            className="property__image"
            src="https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/16.jpg"
          />
        </div>
        <div
          className="property__image-wrapper"
          key="12"
        >
          <img
            alt="Photo studio"
            className="property__image"
            src="https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/14.jpg"
          />
        </div>
        <div
          className="property__image-wrapper"
          key="13"
        >
          <img
            alt="Photo studio"
            className="property__image"
            src="https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/5.jpg"
          />
        </div>
      </div>
    </div>
    <div
      className="property__container container"
    >
      <div
        className="property__wrapper"
      >
        <div
          className="property__name-wrapper"
        >
          <h1
            className="property__name"
          >
            House in countryside
          </h1>
          <button
            className="property__bookmark-button button"
            type="button"
          >
            <svg
              className="property__bookmark-icon"
              height={33}
              width={31}
            >
              <use
                xlinkHref="#icon-bookmark"
              />
            </svg>
            <span
              className="visually-hidden"
            >
              To bookmarks
            </span>
          </button>
        </div>
        <div
          className="property__rating rating"
        >
          <div
            className="property__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "84%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
          <span
            className="property__rating-value rating__value"
          >
            4.8
          </span>
        </div>
        <ul
          className="property__features"
        >
          <li
            className="property__feature property__feature--entire"
          >
            Entire place
          </li>
          <li
            className="property__feature property__feature--bedrooms"
          >
            5
             Bedrooms
          </li>
          <li
            className="property__feature property__feature--adults"
          >
            Max 
            7
             adults
          </li>
        </ul>
        <div
          className="property__price"
        >
          <b
            className="property__price-value"
          >
            €
            656
          </b>
          <span
            className="property__price-text"
          >
             night
          </span>
        </div>
        <div
          className="property__inside"
        >
          <h2
            className="property__inside-title"
          >
            What's inside
          </h2>
          <ul
            className="property__inside-list"
          >
            <li
              className="property__inside-item"
              key="0"
            >
              Laptop friendly workspace
            </li>
            <li
              className="property__inside-item"
              key="1"
            >
              Washer
            </li>
            <li
              className="property__inside-item"
              key="2"
            >
              Air conditioning
            </li>
            <li
              className="property__inside-item"
              key="3"
            >
              Breakfast
            </li>
            <li
              className="property__inside-item"
              key="4"
            >
              Towels
            </li>
            <li
              className="property__inside-item"
              key="5"
            >
              Baby seat
            </li>
          </ul>
        </div>
        <div
          className="property__host"
        >
          <h2
            className="property__host-title"
          >
            Meet the host
          </h2>
          <div
            className="property__host-user user"
          >
            <div
              className="property__avatar-wrapper property__avatar-wrapper--pro user__avatar-wrapper"
            >
              <img
                alt="Host avatar"
                className="property__avatar user__avatar"
                height={74}
                src="img/avatar-angelina.jpg"
                width={74}
              />
            </div>
            <span
              className="property__user-name"
            >
              Angelina
            </span>
            <span
              className="property__user-status"
            >
              Pro
            </span>
          </div>
          <div
            className="property__description"
          >
            <p
              className="property__text"
            >
              Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.
            </p>
          </div>
        </div>
        <section
          className="property__reviews reviews"
        >
          <h2
            className="reviews__title"
          >
            Reviews · 
            <span
              className="reviews__amount"
            >
              2
            </span>
          </h2>
          <ReviewsList
            reviews={
              Array [
                Object {
                  "avatar": "img/avatar-max.jpg",
                  "dateString": "April 2019",
                  "dateTime": "2019-04-24",
                  "id": 1,
                  "message": "A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.
              The building is green and from 18th century.",
                  "rating": 4.9,
                  "userName": "Max",
                },
                Object {
                  "avatar": "img/avatar-angelina.jpg",
                  "dateString": "Mart 2019",
                  "dateTime": "2019-03-20",
                  "id": 2,
                  "message": "A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.
              The building is green and from 18th century.",
                  "rating": 4.5,
                  "userName": "Dasha",
                },
              ]
            }
          />
          <form
            action="#"
            className="reviews__form form"
            method="post"
          >
            <label
              className="reviews__label form__label"
              htmlFor="review"
            >
              Your review
            </label>
            <div
              className="reviews__rating-form form__rating"
            >
              <input
                className="form__rating-input visually-hidden"
                defaultValue={5}
                id="5-stars"
                name="rating"
                type="radio"
              />
              <label
                className="reviews__rating-label form__rating-label"
                htmlFor="5-stars"
                title="perfect"
              >
                <svg
                  className="form__star-image"
                  height={33}
                  width={37}
                >
                  <use
                    xlinkHref="#icon-star"
                  />
                </svg>
              </label>
              <input
                className="form__rating-input visually-hidden"
                defaultValue={4}
                id="4-stars"
                name="rating"
                type="radio"
              />
              <label
                className="reviews__rating-label form__rating-label"
                htmlFor="4-stars"
                title="good"
              >
                <svg
                  className="form__star-image"
                  height={33}
                  width={37}
                >
                  <use
                    xlinkHref="#icon-star"
                  />
                </svg>
              </label>
              <input
                className="form__rating-input visually-hidden"
                defaultValue={3}
                id="3-stars"
                name="rating"
                type="radio"
              />
              <label
                className="reviews__rating-label form__rating-label"
                htmlFor="3-stars"
                title="not bad"
              >
                <svg
                  className="form__star-image"
                  height={33}
                  width={37}
                >
                  <use
                    xlinkHref="#icon-star"
                  />
                </svg>
              </label>
              <input
                className="form__rating-input visually-hidden"
                defaultValue={2}
                id="2-stars"
                name="rating"
                type="radio"
              />
              <label
                className="reviews__rating-label form__rating-label"
                htmlFor="2-stars"
                title="badly"
              >
                <svg
                  className="form__star-image"
                  height={33}
                  width={37}
                >
                  <use
                    xlinkHref="#icon-star"
                  />
                </svg>
              </label>
              <input
                className="form__rating-input visually-hidden"
                defaultValue={1}
                id="1-star"
                name="rating"
                type="radio"
              />
              <label
                className="reviews__rating-label form__rating-label"
                htmlFor="1-star"
                title="terribly"
              >
                <svg
                  className="form__star-image"
                  height={33}
                  width={37}
                >
                  <use
                    xlinkHref="#icon-star"
                  />
                </svg>
              </label>
            </div>
            <textarea
              className="reviews__textarea form__textarea"
              defaultValue=""
              id="review"
              name="review"
              placeholder="Tell how was your stay, what you like and what can be improved"
            />
            <div
              className="reviews__button-wrapper"
            >
              <p
                className="reviews__help"
              >
                To submit review please make sure to set 
                <span
                  className="reviews__star"
                >
                  rating
                </span>
                 and describe your stay with at least 
                <b
                  className="reviews__text-amount"
                >
                  50 characters
                </b>
                .
              </p>
              <button
                className="reviews__submit form__submit button"
                disabled={true}
                type="submit"
              >
                Submit
              </button>
            </div>
          </form>
        </section>
      </div>
    </div>
    <section
      className="property__map map"
    >
      <OffersMap
        offers={
          Array [
            Object {
              "bedrooms": 3,
              "city": Object {
                "location": Object {
                  "latitude": 52.370216,
                  "longitude": 4.895168,
                  "zoom": 10,
                },
                "name": "Amsterdam",
              },
              "description": "A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.",
              "goods": Array [
                "Heating",
                "Kitchen",
                "Cable TV",
                "Washing machine",
                "Coffee machine",
                "Dishwasher",
              ],
              "host": Object {
                "avatarUrl": "img/avatar-angelina.jpg",
                "id": 3,
                "isPro": true,
                "name": "Angelina",
              },
              "id": 1,
              "images": Array [
                "/img/room.jpg",
                "img/apartment-01.jpg",
                "img/apartment-02.jpg",
                "img/apartment-03.jpg",
                "img/studio-01.jpg",
                "img/room.jpg",
              ],
              "isFavorite": true,
              "isPremium": true,
              "location": Object {
                "latitude": 52.3909553943508,
                "longitude": 4.85309666406198,
                "zoom": 8,
              },
              "maxAdults": 4,
              "previewImage": "img/apartment-01.jpg",
              "price": 120,
              "rating": 4.8,
              "title": "Beautiful & luxurious studio at great location",
              "type": "Apartment",
            },
            Object {
              "bedrooms": 1,
              "city": Object {
                "location": Object {
                  "latitude": 52.370216,
                  "longitude": 4.895168,
                  "zoom": 10,
                },
                "name": "Amsterdam",
              },
              "description": "A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.",
              "goods": Array [
                "Heating",
                "Kitchen",
                "Cable TV",
                "Washing machine",
                "Coffee machine",
                "Dishwasher",
              ],
              "host": Object {
                "avatarUrl": "img/avatar-angelina.jpg",
                "id": 3,
                "isPro": true,
                "name": "Angelina",
              },
              "id": 2,
              "images": Array [
                "/img/room.jpg",
                "img/apartment-01.jpg",
                "img/apartment-02.jpg",
                "img/apartment-03.jpg",
                "img/studio-01.jpg",
                "img/room.jpg",
              ],
              "isFavorite": false,
              "isPremium": false,
              "location": Object {
                "latitude": 52.369553943508,
                "longitude": 4.85309666406198,
                "zoom": 8,
              },
              "maxAdults": 2,
              "previewImage": "img/room.jpg",
              "price": 80,
              "rating": 4.2,
              "title": "Wood and stone place",
              "type": "Private room",
            },
            Object {
              "bedrooms": 2,
              "city": Object {
                "location": Object {
                  "latitude": 52.370216,
                  "longitude": 4.895168,
                  "zoom": 10,
                },
                "name": "Amsterdam",
              },
              "description": "A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.",
              "goods": Array [
                "Heating",
                "Kitchen",
                "Cable TV",
                "Washing machine",
                "Coffee machine",
                "Dishwasher",
              ],
              "host": Object {
                "avatarUrl": "img/avatar-angelina.jpg",
                "id": 3,
                "isPro": true,
                "name": "Angelina",
              },
              "id": 3,
              "images": Array [
                "/img/room.jpg",
                "img/apartment-01.jpg",
                "img/apartment-02.jpg",
                "img/apartment-03.jpg",
                "img/studio-01.jpg",
                "img/room.jpg",
              ],
              "isFavorite": true,
              "isPremium": false,
              "location": Object {
                "latitude": 52.3909553943508,
                "longitude": 4.929309666406198,
                "zoom": 8,
              },
              "maxAdults": 3,
              "previewImage": "img/apartment-02.jpg",
              "price": 70,
              "rating": 4,
              "title": "Nice, cozy, warm big bed apartmen",
              "type": "Apartment",
            },
          ]
        }
        onHoverOfferId={-1}
        selectedCity={
          Object {
            "location": Object {
              "latitude": 52.37454,
              "longitude": 4.897976,
              "zoom": 13,
            },
            "name": "Amsterdam",
          }
        }
      />
    </section>
  </section>
  <div
    className="container"
  >
    <section
      className="near-places places"
    >
      <h2
        className="near-places__title"
      >
        Other places in the neighbourhood
      </h2>
      <OffersList
        classNames="near-places__list"
        offers={
          Array [
            Object {
              "bedrooms": 3,
              "city": Object {
                "location": Object {
                  "latitude": 52.370216,
                  "longitude": 4.895168,
                  "zoom": 10,
                },
                "name": "Amsterdam",
              },
              "description": "A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.",
              "goods": Array [
                "Heating",
                "Kitchen",
                "Cable TV",
                "Washing machine",
                "Coffee machine",
                "Dishwasher",
              ],
              "host": Object {
                "avatarUrl": "img/avatar-angelina.jpg",
                "id": 3,
                "isPro": true,
                "name": "Angelina",
              },
              "id": 1,
              "images": Array [
                "/img/room.jpg",
                "img/apartment-01.jpg",
                "img/apartment-02.jpg",
                "img/apartment-03.jpg",
                "img/studio-01.jpg",
                "img/room.jpg",
              ],
              "isFavorite": true,
              "isPremium": true,
              "location": Object {
                "latitude": 52.3909553943508,
                "longitude": 4.85309666406198,
                "zoom": 8,
              },
              "maxAdults": 4,
              "previewImage": "img/apartment-01.jpg",
              "price": 120,
              "rating": 4.8,
              "title": "Beautiful & luxurious studio at great location",
              "type": "Apartment",
            },
            Object {
              "bedrooms": 1,
              "city": Object {
                "location": Object {
                  "latitude": 52.370216,
                  "longitude": 4.895168,
                  "zoom": 10,
                },
                "name": "Amsterdam",
              },
              "description": "A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.",
              "goods": Array [
                "Heating",
                "Kitchen",
                "Cable TV",
                "Washing machine",
                "Coffee machine",
                "Dishwasher",
              ],
              "host": Object {
                "avatarUrl": "img/avatar-angelina.jpg",
                "id": 3,
                "isPro": true,
                "name": "Angelina",
              },
              "id": 2,
              "images": Array [
                "/img/room.jpg",
                "img/apartment-01.jpg",
                "img/apartment-02.jpg",
                "img/apartment-03.jpg",
                "img/studio-01.jpg",
                "img/room.jpg",
              ],
              "isFavorite": false,
              "isPremium": false,
              "location": Object {
                "latitude": 52.369553943508,
                "longitude": 4.85309666406198,
                "zoom": 8,
              },
              "maxAdults": 2,
              "previewImage": "img/room.jpg",
              "price": 80,
              "rating": 4.2,
              "title": "Wood and stone place",
              "type": "Private room",
            },
            Object {
              "bedrooms": 2,
              "city": Object {
                "location": Object {
                  "latitude": 52.370216,
                  "longitude": 4.895168,
                  "zoom": 10,
                },
                "name": "Amsterdam",
              },
              "description": "A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.",
              "goods": Array [
                "Heating",
                "Kitchen",
                "Cable TV",
                "Washing machine",
                "Coffee machine",
                "Dishwasher",
              ],
              "host": Object {
                "avatarUrl": "img/avatar-angelina.jpg",
                "id": 3,
                "isPro": true,
                "name": "Angelina",
              },
              "id": 3,
              "images": Array [
                "/img/room.jpg",
                "img/apartment-01.jpg",
                "img/apartment-02.jpg",
                "img/apartment-03.jpg",
                "img/studio-01.jpg",
                "img/room.jpg",
              ],
              "isFavorite": true,
              "isPremium": false,
              "location": Object {
                "latitude": 52.3909553943508,
                "longitude": 4.929309666406198,
                "zoom": 8,
              },
              "maxAdults": 3,
              "previewImage": "img/apartment-02.jpg",
              "price": 70,
              "rating": 4,
              "title": "Nice, cozy, warm big bed apartmen",
              "type": "Apartment",
            },
          ]
        }
      />
    </section>
  </div>
</main>
`;
